src/main/scala/lms/core/stub.scala

```diff
@@ -840,6 +840,15 @@ trait PrimitiveOps extends Base with OverloadHack {
     def toLong(implicit __pos: SourceContext): Rep[Long] = cast_helper[Char,Long](self)
   }
 
+  // FIMXE(feiw) this overlaps with the implicit class of specific types
+  implicit class GenericMathOpsCls[T:Numeric:Manifest](val self: Rep[T])(implicit __pos: SourceContext) {
```

Comment:
So at later point, other classes for numeric syntax (like `PrimitiveMathOpsIntOpsCls` and `PrimitiveMathOpsLongOpsCls`) can be subsumed by `GenericMathOpsCls`? But it seems we still need to have a lot of boilerplate code when the operands have different types.

Reply:
yeah, I don't know how to do so for different types.